[CSS] Refactor variant css tokens

rav3 opened the pull request:
## :open_book: Description

Refactor variant css tokens
Adding `$theme-colors-rgb` color mixin loop 
Generating all $theme-colors in `variables.scss`.

Both badge and spinner components are working as intended.

```
Please delete options that are not relevant.
```

- [x] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)

## :test_tube: How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- [x] Tested in storybook visually. 

## :white_check_mark: Checklist:

- [x] My code follows the SGDS style guidelines and naming conventions
- [x]  I have performed a self-review of my own code
- [x]  I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] Any dependent changes have been merged and published in downstream modules

clukhei reviewed (approved):
THank you its working great

rav3 commented:
> THank you its working great

Forgot to mention this PR includes the badge fixes as well.
- Added stories to the badge
- Fixed `isLight` variant by generating `--sgds-*state-color*` custom variables in `sgds-elements.scss`
